Leetcode/529-Minesweeper.py

```diff
@@ -0,0 +1,33 @@
+from collections import deque
+class Solution:
+    def updateBoard(self, board: List[List[str]], click: List[int]) -> List[List[str]]:
+        dirs = [(0, 1), (1, 0), (1, 1), (-1, -1), (0, -1), (-1, 0), (1, -1), (-1, 1)]
+        m, n = len(board), len(board[0])
+
+        queue = deque([click])
+        visited = set([click[0], click[1]])
+        while queue:
+            r, c = queue.popleft()
+            visited.add((r, c))
+
+            if board[r][c] == 'M':
+                board[r][c] = 'X'
+                break
+            mines, toreveal = 0, []
```

Comment:
why not just use the `queue.append((nr,nc))` with `if board[r][c] == 'E':` condition instead of using extra space?